src/pkt_proc.c

```diff
@@ -984,20 +1082,39 @@ void* process_packet (unsigned char *ctx_ptr,
         gettimeofday(&now,NULL);
         dyn_header->ts.tv_sec = now.tv_sec;
         dyn_header->ts.tv_usec = now.tv_usec;
-        dyn_header->caplen = ip->ip_len;
-        dyn_header->len = ip->ip_len;
+        if (ctx->curr_pkt_type == ETH_TYPE_IPV6) {
+            dyn_header->caplen = ipv6->ip_len;
+            dyn_header->len = ipv6->ip_len;
+        } else {
+            dyn_header->caplen = ip->ip_len;
+            dyn_header->len = ip->ip_len;
+        }
         header = dyn_header;
     }
 
-    ip_len = ntohs(ip->ip_len);
-    if (ip_len < sizeof(struct ip_hdr)) {
-        /*
-         * IP packet is malformed (shorter than a complete IP header)
-         */
-        if (allocated_packet_header)
-            free(dyn_header);
-        return NULL;
+    if (ctx->curr_pkt_type == ETH_TYPE_IPV6) {
+        ip_len = ntohs(ipv6->ip_len);
+        if (header->caplen < IPV6_HDR_LENGTH) {
+            /*
+             * IP packet is malformed shorter than a complete IP header
+             */
+            if (allocated_packet_header)
+                free(dyn_header);
```

Comment:
Its good practice to set object ptr to NULL after free. Your call.

Reply:
this is a local variable that is set to NULL when the function is entered. When the memory is being free'd, the function is exiting, so it will be set to NULL the next time the function is called.